### ISSUE TITLE
SMTP Notifications setup problem

### ISSUE THREAD

**Issue by TheAnachronism** [louislam/uptime-kuma#690]
**Describe your problem**
Setup notifications with smtp not working with selfhosted mailcow mail-server.
I have the correct hostname, port and authentication settings because I tested it with a different software for mail notifications.
What I am curios about is, that in the logs it has `connect ECONNREFUSED XXX.XXX.XXX.XXX:465` and not the actual hostname with domain. Maybe I understand SMTP or mail wrong but I thought it requires the use of the hostname for secure connections.
In the working version in the other software it won't work with the ip address so I though maybe that's the problem.

**Info**
Uptime Kuma Version: latest
Using Docker?: Yes
Docker Version:20.10.9
OS: ubuntu-21-04

**Comment by TheAnachronism** [comment]
About my config:
@c0by85 
![image](https://user-images.githubusercontent.com/32616088/137124312-88948942-799e-43c3-9c50-9c858dd4d96f.png)

Do I maybe have something wrong with my mailcow config?
Mailcow is running as docker containers inside an ubuntu VM
Port 465 is forwarded correctly

**Comment by TheAnachronism** [comment]
Even with STARTTLS 587 it doesn't work.
I think I messed something up somewhere with my mail config in mailcow or so but I'm still very confused on why it works with my game panel (pterodactyl)

**Comment by louislam** [comment]
> Even with STARTTLS 587 it doesn't work. I think I messed something up somewhere with my mail config in mailcow or so but I'm still very confused on why it works with my game panel (pterodactyl)

I think you can set a TCP monitor, see if Uptime Kuma can connect to your mail server first.

**Comment by TheAnachronism** [comment]
Also I discovered that with the TCP monitor...

**Comment by louislam** [comment]
Btw why pterodactyl is working? Maybe put Uptime Kuma's container next to it?

**Comment by MrEddX** [comment]
> If its not for human are there other firewalls that correctly work with the entire proxmox stuff?

Proxmox has built in firewall that can be managed on different levels. Datacenter -> Node -> VM/CT. The GUI makes is super easy to manage all the rules. Templates are also available for most common services as SSH, HTTP, ICMP, and so on.

You can find simple, but in the same time good explanation with examples in this [video](https://youtu.be/DNsLLrCgK0U).

**Comment by TheAnachronism** [comment]
Alright so I found what the problem with everything was. My mailserver is running inside a VM as a docker container as well as uptime-kuma. I have forwarded the SMTP ports (465 and 587) to the VM with firewalld. The problem is, that I only did that in the external zone of firewalld which has my public interface assigned. But the network interface for the VM is inside a different zone, which doesn't have that forwarding rule. Because of that the port forward didn't work for requests coming from inside the VM network but only for completely external requests.
As soon as I added the same rule to the other zone everything worked.
Now just a weird issue is, that all requests that want to use that port coming from inside that network are forwarded to the VM. Even those that aren't targeted at the public ip address of my server. But that is an issue with firewalld and has nothing to do with this so I'll solve it somewhere else but if anyone is interested, here is a serverfault question I made about it:
https://serverfault.com/questions/1080546/firewalld-port-forwarding-proxmox-makes-port-unusable-for-other-connections
